crates/integration_test/src/logger.rs

```diff
@@ -0,0 +1,114 @@
+use anyhow::{Context, Result};
+use log::{LevelFilter, Log, Metadata, Record};
+use once_cell::sync::OnceCell;
+use std::borrow::Cow;
+use std::fs::{File, OpenOptions};
+use std::io::{stderr, Write};
+use std::path::PathBuf;
+use std::str::FromStr;
+
+pub static LOG_FILE: OnceCell<Option<File>> = OnceCell::new();
+const LOG_LEVEL_ENV_NAME: &str = "YOUKI_INTEGRATION_LOG_LEVEL";
+
+/// Initialize the logger, must be called before accessing the logger
+/// Multiple parts might call this at once, but the actual initialization
+/// is done only once due to use of OnceCell
+pub fn init(log_file: Option<PathBuf>, debug: bool) -> Result<()> {
```

Comment:
I think it is better that the output of the integration test should be handled by only stderr/stdout. If you want to write to the file, you can use redirect, tee, etc